File: app/src/main/java/com/example/pma_project/DataAdapter.java
package com.example.pma_project;

import android.content.Intent;
import android.graphics.BitmapFactory;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class DataAdapter extends RecyclerView.Adapter<DataAdapter.ViewHolder> {

    private ArrayList<Data> dataEntries;

    public DataAdapter(ArrayList<Data> dataEntries){
        this.dataEntries = dataEntries;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
    {
        View v = (View) LayoutInflater.from(parent.getContext()).inflate(R.layout.item_history, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Data dataToShow = dataEntries.get(position);

        holder.title.setText(dataToShow.getTitle());
        holder.date.setText(dataToShow.getDate());

        holder.image.setImageBitmap(BitmapFactory.decodeFile(dataToShow.getPathToImage()));
//        holder.itemView.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Toast.makeText(v.getContext(), "Recycle Click" + position, Toast.LENGTH_LONG).show();
//            }
//        });
    }

    @Override
    public int getItemCount() {
        if (dataEntries != null) {
            return dataEntries.size();
        } else {
            return 0;
        }
    }

    public static class ViewHolder extends RecyclerView.ViewHolder{
        public final View view;
        public final TextView title;
        public final TextView date;
        public final ImageView image;


        public ViewHolder(@NonNull View view) {
            super(view);
            this.view = view;
            title = view.findViewById(R.id.titleHistoryPage);
            date = view.findViewById(R.id.dateHistoryPage);
            image = view.findViewById(R.id.imageHistoryPage);
            view.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent intent = new Intent(v.getContext(),MainActivity.class);
                    intent.putExtra("position", getAdapterPosition());
                    v.getContext().startActivity(intent);
                }
            });
        }
    }
}

File: app/src/main/java/com/example/pma_project/DataAdapter.kt
package com.example.pma_project

import android.content.Intent
import android.graphics.BitmapFactory
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.pma_project.MainActivity
import java.util.*

class DataAdapter(private val dataEntries: ArrayList<Data>?) : RecyclerView.Adapter<DataAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_history, parent, false) as View
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val dataToShow = dataEntries!![position]
        holder.title.text = dataToShow.title
        holder.date.text = dataToShow.date
        holder.image.setImageBitmap(BitmapFactory.decodeFile(dataToShow.pathToImage))
        //        holder.itemView.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Toast.makeText(v.getContext(), "Recycle Click" + position, Toast.LENGTH_LONG).show();
//            }
//        });
    }

    override fun getItemCount(): Int {
        return dataEntries?.size ?: 0
    }

    class ViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
        val title: TextView = view.findViewById(R.id.titleHistoryPage)
        val date: TextView = view.findViewById(R.id.dateHistoryPage)
        val image: ImageView = view.findViewById(R.id.imageHistoryPage)

        init {
            view.setOnClickListener { v ->
                val intent = Intent(v.context, MainActivity::class.java)
                intent.putExtra("position", adapterPosition)
                v.context.startActivity(intent)
            }
        }
    }
}